src/main/java/com/bootcamp/demo/restapi/controller/FirebaseController.java

```diff
@@ -1,54 +1,40 @@
-package com.bootcamp.demo;
+package com.bootcamp.demo.restapi.controller;
 
 import com.google.cloud.firestore.CollectionReference;
 import com.google.cloud.firestore.Firestore;
-import com.google.firebase.FirebaseOptions;
-import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 import org.springframework.web.bind.annotation.GetMapping;
 import org.springframework.web.bind.annotation.RequestMapping;
 import org.springframework.web.bind.annotation.RestController;
 
-import javax.annotation.PostConstruct;
-import java.io.ByteArrayInputStream;
-import java.io.IOException;
 import java.util.Set;
 import java.util.stream.Collectors;
 import java.util.stream.StreamSupport;
 
-import static com.google.auth.oauth2.GoogleCredentials.fromStream;
-import static com.google.firebase.FirebaseApp.initializeApp;
-import static com.google.firebase.cloud.FirestoreClient.getFirestore;
-import static java.lang.System.getProperty;
-import static java.nio.charset.StandardCharsets.UTF_8;
 import static org.springframework.http.MediaType.APPLICATION_JSON_VALUE;
 
 /**
  * Sample RestController
  * Demo - Used for testing purposes
  */
-@ConditionalOnProperty("firebaseKey")
 @RestController
 @RequestMapping(path = "/firebase", produces = APPLICATION_JSON_VALUE)
 public class FirebaseController {
-    private Firestore firestoreDB;
 
-    @PostConstruct
-    private void initFirestore() throws IOException {
-        final var serviceAccount = new ByteArrayInputStream(getProperty("firebaseKey").getBytes(UTF_8));
+    private static final Logger LOGGER = LoggerFactory.getLogger(FirebaseController.class);
+    private final Firestore firestoreDB;
 
-        final var options = FirebaseOptions.builder()
-                .setCredentials(fromStream(serviceAccount))
-                .build();
-        initializeApp(options);
-
-        firestoreDB = getFirestore();
+    private FirebaseController(Firestore firestoreDB) {
+        this.firestoreDB = firestoreDB;
     }
 
     /**
      * Returns the path for all collections stored in Firestore
      */
     @GetMapping("/getAllPaths")
     public Set<String> getAllPaths() {
+        LOGGER.info("GET FIREBASE COLLECTIONS - API endpoint invoked");
```

Comment:
These types of logs are redundant.
Often logging too much at info level results in logs pollution. Logging should be used as long as it is helpful but it is not supposed to be a "finer" form of debugging.
Applications can have several types of logging files i.e. access logs/server logs and the purpose for each is different.
An access log will keep track of all invoked endpoints, whereas in the server log you usually add logs to pick up when something does not work as expected (exceptional/error flows etc.).
You can find a basic intro here: https://www.baeldung.com/spring-boot-embedded-tomcat-logs